staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -352,7 +360,20 @@ func (s preparedGenericAPIServer) Run(stopCh <-chan struct{}) error {
 	}()
 
 	// close socket after delayed stopCh
-	stoppedCh, listenerStoppedCh, err := s.NonBlockingRun(delayedStopCh.Signaled())
+	drainedCh := s.lifecycleSignals.InFlightRequestsDrained
+	stopHttpServerCh := delayedStopCh.Signaled()
+	shutdownTimeout := s.ShutdownTimeout
+	if s.KeepListeningDuringGracefulTermination {
+		// when this mode is enabled, we do the following:
+		// - the server will continue to listen until all existing requests in flight
+		//   (not including active long runnning requests) have been drained.
+		// - once drained, http Server Shutdown is invoked with a timeout of 5s
+		stopHttpServerCh = drainedCh.Signaled()
+		shutdownTimeout = 2 * time.Second
+		klog.V(1).InfoS("[graceful-termination] HTTP Server", "ShutdownTimeout", shutdownTimeout)
```

Comment:
The `[graceful-termination]` prefix is used everywhere?

Reply:
yes, so we can search with it during debugging

Reply:
I mean the syntax. It looks special. `Using graceful termination http server shutdown timeout` would be more normal.